### PR TITLE
Fix #2780

### LINKED ISSUE #2780: Add havecode field in `i` list
Only true if info->arch is != NULL

### PR DESCRIPTION
**Pull request by alvarofe**
is this ok? tomorrow i'll update the test to include this new field

### CODE REVIEW

**Comment by Maijin** [comment]
I hate those tests x) Always need to be fixed :P

File: libr/core/bin.c [hunk 1]
```diff
@@ -394,20 +394,36 @@ static const char* get_compile_time(Sdb *binFileSdb) {
 	return timeDateStamp_string;
 }
 
+static int is_executable (RBinObject *obj) {
+	RListIter *it;
+	RBinSection* sec;
+	if (obj) {
+		if (obj->info && obj->info->arch)
```
Comment (radare) [inline libr/core/bin.c]: not sure if this case will be 100% true and maybe we want to check entrypoints too

**Comment by radare** [comment]
lets wait to review all travis results and see if we have any false positive.

**Comment by radare** [comment]
Maybe checking for entrypoints is not necessary because a library can have no entrypoints and code

File: libr/core/bin.c [hunk 2]
```diff
@@ -394,20 +394,36 @@ static const char* get_compile_time(Sdb *binFileSdb) {
...
 	const char *compiled = NULL;
+	bool havecode;
 
-	if (!binfile || !info) {
+	if (!binfile || !info || !obj) {
 		if (mode & R_CORE_BIN_JSON) r_cons_printf ("{}");
 		return false;
 	}
-
+	havecode = is_executable (obj) | (obj->entries != NULL);
```
Comment (alvarofe) [inline libr/core/bin.c]: that is why here i check if the file contains executable section or it has an entry point

**Comment by radare** [comment]
Oh didnt saw that. lgtm, let's see how many false positives we see :D -.. merging!

**Pull request closed without merging**